frontend/packages/console-dynamic-plugin-sdk/scripts/utils/typescript.ts

```diff
@@ -22,16 +22,21 @@ export const getSchemaGeneratorConfig = (srcFile: string, typeName?: string): ts
 });
 
 export const getProgram = (config: tsj.Config): ts.Program => {
-  const program = tsj.createProgram(config);
+  const program = tsj.createProgram({ ...tsj.DEFAULT_CONFIG, ...config });
 
   const diagnostics = ts.sortAndDeduplicateDiagnostics(ts.getPreEmitDiagnostics(program));
   if (diagnostics.length > 0) {
     console.error(formatDiagnostics(diagnostics, program.getCurrentDirectory()));
   }
 
-  const hasDiagnosticErrors = diagnostics.some((d) => d.category === ts.DiagnosticCategory.Error);
-  if (hasDiagnosticErrors) {
-    throw new Error(`Detected errors while parsing ${relativePath(config.path)}`);
+  const diagnosticsErrors = diagnostics.filter((d) => d.category === ts.DiagnosticCategory.Error);
+
+  if (diagnosticsErrors.length > 0) {
+    throw new Error(
+      `Detected errors while parsing ${relativePath(config.path ?? 'unknown')}, ${String(
```

Comment:
Just wondering, why do we need the `?? 'unknown'` fallback?

Reply:
The type for `config.path` is actually `string | undefined` but `relativePath` cannot except `undefined`

Reply:
I think this is one of those cases when TS thinks the value may be undefined but in practice it should never be undefined.

I'd just use https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html here to keep code simple:
```ts
relativePath(config.path!)
```